sklearn/datasets/_twenty_newsgroups.py

```diff
@@ -433,12 +451,22 @@ def fetch_20newsgroups_vectorized(*, subset="train", remove=(), data_home=None,
                                    download_if_missing=download_if_missing)
 
     if os.path.exists(target_file):
-        X_train, X_test = joblib.load(target_file)
+        try:
+            X_train, X_test, feature_names = joblib.load(target_file)
```

Comment:
Running this locally, forces us to delete `/Users/thomasfan/scikit_learn_data/20newsgroup_vectorized_py3.pkl`.

Would there be a downside with removing the `target_file` when a `ValueError` is raised?

Reply:
It would be fine with me as well. Do you think of a case where trying to delete the file would fail (with some OS lock)?

Reply:
Good point, it would make sense to do something similar then.

Reply:
Actually I am unsure if this is a good idea. We might post-pone this part and come with something consistent across all fetchers.

I am thinking about the following point:

* it might be better to have a download and retry common to all fetcher
* in case `download_if_missing=False` and that we have an invalidated cache, then which type of error do we raise?

Reply:
> We might post-pone this part and come with something consistent across all fetchers.

I think I am okay with this. I hope there isn't any more testing related issues with this PR. Once this gets merged, developers would need to manually remove the pickled file.

> it might be better to have a download and retry common to all fetcher

Yea... it makes sense to do.

> in case download_if_missing=False and that we have an invalidated cache, then which type of error do we raise?

Cache is invalid, please set `download_if_missing=True` to redownload ?

Reply:
OK so we would need to detect that we removed the file before.